ocaml/xapi/repository.ml

```diff
@@ -602,23 +612,34 @@ let apply_updates ~__context ~host ~hash =
     then
       raise Api_errors.(Server_error (updateinfo_hash_mismatch, [])) ;
     with_pool_repositories (fun () ->
-        let all_updates =
+        let host_updates =
           http_get_host_updates_in_json ~__context ~host ~installed:true
+        in
+        let latest_updates =
+          host_updates
           |> Yojson.Basic.Util.member "updates"
           |> Yojson.Basic.Util.to_list
           |> List.map Update.of_json
         in
-        match all_updates with
+        match latest_updates with
```

Comment:
Yes. But there is another case. `latest_updates` is used as the set of updates to be installed in `Client.Client.Repository.apply` actually. It is possible that `accumulative_updates` is empty, but `latest_updates` is not. (This means some packages are missing in updateinfo. But this will not break the actual updates since updateinfo is only information of the updates. In other words, we only try best to retrieve information from `accumulative_updates` )
So here `latest_updates` is the most accurate indicator to determine if a host could be updated.